File: doc/FizzBuzz8.scala
  def addOption(a1: Option[String], a2: Option[String]):
      Option[String] = (a1, a2) match {
    case (Some(s1), None)     => Some(s1)
    case (None,     Some(s2)) => Some(s2)
    case (Some(s1), Some(s2)) => Some(s1 + s2)
    case (None,     None)     => None
  }

File: doc/FizzBuzz8.swift
func addOption(a1: String?, a2: String?) -> String? = {
  switch (a1, a2) {
  case (let .Some(s1), .None):         return .Some(s1)
  case (.None, let .Some(s2)):         return .Some(s2)
  case (let .Some(s1), let .Some(s2)): return .Some(s1+s2)
  case (.None, .None):                 return .None
  }
}
